src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -83,22 +83,54 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
-        public static Dictionary<Run, int> ComputeRunResultCounts(this SarifLog log)
+        public static List<string> GetRunToolNames(this SarifLog log)
         {
-            if (log == null) { throw new ArgumentNullException(nameof(log)); }
-            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+            return log?.Runs?.Where(r => r?.Results?.Count > 0).Select(r => r.Tool?.Driver?.Name).Distinct().ToList();
+        }
 
-            var resultCountsByRun = new Dictionary<Run, int>();
+        public static int GetAggregateResultCount(this SarifLog log)
+        {
+            return log?.Runs?.Select(r => r?.Results?.Count).Sum() ?? 0;
+        }
 
-            foreach (Run run in log?.Runs)
+        public static string PhraseToolNames(List<string> toolNames)
+        {
+            toolNames = toolNames.Where(r => !string.IsNullOrWhiteSpace(r)).ToList();
+            if (toolNames.Count == 0)
+            {
+                return string.Empty;
+            }
+            else if (toolNames.Count == 1)
+            {
+                return string.Format("'{0}'", toolNames[0]);
+            }
+            else
             {
-                if (run?.Results != null)
+                string phrasedToolNames = string.Format("'{0}'", toolNames[0]);
+                for (int j = 1; j < toolNames.Count - 1; j++)
                 {
-                    resultCountsByRun.Add(run, run.Results.Count);
+                    phrasedToolNames = string.Join(", ", phrasedToolNames, string.Format("'{0}'", toolNames[j]));
                 }
+                phrasedToolNames = string.Join(" and ", phrasedToolNames, string.Format("'{0}'", toolNames[toolNames.Count - 1]));
+                return phrasedToolNames;
             }
+        }
+
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
+        {
+            List<string> toolNames = log.GetRunToolNames();
+            int totalResults = log.GetAggregateResultCount();
+            string phrasedToolNames = PhraseToolNames(toolNames);
```

Comment:
>string [](start = 12, length = 6)

Here is a nit so small you'll need an electron microscope to see it, yet IMO it's the sort of tiny readability improvement that, when practiced uniformly in the code base, makes a real difference.

On line 121 you assign to `toolNames`, on line 122 you do something unrelated, and on line 123 you _use_ `toolNames`. There's a readability principle that says to limit variable scope as much as possible, because that _minimizes how much stuff you have to remember/keep in mind_.

So if you you reverse the order of lines 121 and 122, you don't introduce `toolName` until you actually need it, and you don't have to remember that it was initialized somewhere above. Sure, here it's just one line of separation, but sometimes it's more. And if you always wait to define something until you need it, you'll reduce the "cognitive load" on your readers.